pytorch_lightning/trainer/training_loop.py

```diff
@@ -304,86 +297,92 @@ def process_output(self, output, train):
         # this is just empty shell for code from other class
         pass
 
+    def _training_epoch(self, epoch: int):
+        """Run single training epoch with `epoch` index."""
+        # set seed for distributed sampler (enables shuffling for each epoch)
+        if (self.use_ddp or self.use_tpu) \
+                and hasattr(self.get_train_dataloader().sampler, 'set_epoch'):
+            self.get_train_dataloader().sampler.set_epoch(epoch)
+
+        # get model
+        model = self.get_model()
+
+        # update training progress in trainer and model
+        model.current_epoch = epoch
+        self.current_epoch = epoch
+
+        total_val_batches = 0
+        is_val_epoch = False
+        if not self.disable_validation:
+            # val can be checked multiple times in epoch
+            is_val_epoch = (self.current_epoch + 1) % self.check_val_every_n_epoch == 0
+            val_checks_per_epoch = self.num_training_batches // self.val_check_batch
+            val_checks_per_epoch = val_checks_per_epoch if is_val_epoch else 0
+            total_val_batches = self.num_val_batches * val_checks_per_epoch
+
+        # total batches includes multiple val checks
+        self.total_batches = self.num_training_batches + total_val_batches
+        self.batch_loss_value = 0  # accumulated grads
+
+        if self.fast_dev_run:
+            # limit the number of batches to 2 (1 train and 1 val) in fast_dev_run
+            num_iterations = 2
+        elif self.is_iterable_train_dataloader:
+            # for iterable train loader, the progress bar never ends
+            num_iterations = None
+        else:
+            num_iterations = self.total_batches
+
+        # reset progress bar
+        # .reset() doesn't work on disabled progress bar so we should check
+        if not self.main_progress_bar.disable:
+            self.main_progress_bar.reset(num_iterations)
+        desc = f"Epoch {epoch + 1}" if not self.is_iterable_train_dataloader else ""
+        self.main_progress_bar.set_description(desc)
+
+        # changing gradient according accumulation_scheduler
+        self.accumulation_scheduler.on_epoch_begin()
+
+        # -----------------
+        # RUN TRAINING EPOCH
+        # -----------------
+        self.run_training_epoch()
+
+        # update LR schedulers
+        if self.lr_schedulers:
+            for lr_scheduler in self.lr_schedulers:
+                # accessing the Scheduler state directly,
+                #  see the base class `torch.optim.lr_scheduler._LRScheduler`
+                lr_scheduler.last_epoch = self.current_epoch
```

Comment:
Not sure this is a good idea, any alternatives that involve just calling `.step`?

Reply:
well I can make a loop over and call `.step()` until the `.last_epoch` is the `current_epoch - 1`